pallets/auctions/src/tests.rs

```diff
@@ -0,0 +1,788 @@
+#[cfg(test)]
+use super::mock::*;
+use crate::{mock, Auctions as AuctionsStorage, Error};
+use frame_support::error::BadOrigin;
+use frame_support::{assert_noop, assert_ok};
+use frame_system::RawOrigin;
+use ternoa_common::traits::{MarketplaceTrait, NFTTrait};
+use ternoa_marketplace::Error as MarketplaceError;
+use ternoa_primitives::{
+    marketplace::{MarketplaceId, MarketplaceType},
+    nfts::NFTId,
+};
+
+pub const MIN_AUCTION_DURATION: u64 = 14400;
+// min auction buffer of 1 hour (1*60*60)/6
+pub const MIN_AUCTION_BUFFER: u64 = 600;
+// max auction duration of 30 days (30*24*60*60)/6
+pub const MAX_AUCTION_DURATION: u64 = 432000;
+// auction grace period of 10min (10*60)/6
+pub const AUCTION_GRACE_PERIOD: u64 = 100;
+
+fn get_marketplace(owner: u64) -> MarketplaceId {
+    let owner_signed: mock::Origin = RawOrigin::Signed(owner).into();
+    help::create_mkp(
+        owner_signed,
+        MarketplaceType::Public,
+        10, // 10% commission
+        vec![1],
+        vec![],
+    )
+}
+
+fn create_nft(owner: u64) -> NFTId {
+    let series_id = vec![50];
+    <NFTs as NFTTrait>::create_nft(owner, vec![50], Some(series_id.clone())).unwrap()
+}
+
+fn create_auction(owner: mock::Origin, marketplace_id: MarketplaceId, nft_id: NFTId) {
+    assert_ok!(Auctions::create_auction(
+        owner,
+        nft_id,
+        marketplace_id,
+        MIN_AUCTION_BUFFER + 1,
+        MIN_AUCTION_BUFFER + MIN_AUCTION_DURATION + 1,
+        100,
+        Some(200)
+    ));
+}
+
+#[test]
+fn create_auction_happy() {
+    ExtBuilder::default()
+        .caps(vec![(ALICE, 1000), (BOB, 1000)])
+        .build()
+        .execute_with(|| {
+            let alice: mock::Origin = RawOrigin::Signed(ALICE).into();
+
+            let mkp_id = get_marketplace(ALICE);
+            let nft_id = create_nft(ALICE);
+            create_auction(alice, mkp_id, nft_id);
+
+            // ensure nft is marked as listed for sale
+            assert_eq!(NFTs::is_listed_for_sale(nft_id), Some(true));
```

Comment:
You also need to make sure if the internal state of the auction pallet is correct. The Auctions storage needs to have one entry and that entry needs to have the right details.